test/test_src/ekka_helper_tab.erl

```diff
@@ -0,0 +1,64 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% This module contains definitions that are used for working with the
+%% special marker tab that we're using for storing test metadata.
+-module(ekka_helper_tab).
+
+-export([ mnesia/1
+        , wait_full_replication/1
+        , wait_full_replication/2
+        ]).
+
+-define(TABLE, ?MODULE).
+
+-boot_mnesia({mnesia, [boot]}).
+-copy_mnesia({mnesia, [copy]}).
+-rlog_shard({test_shard, ?TABLE}).
+
+-record(?TABLE, {key, val}).
+
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+mnesia(boot) ->
+    ok = ekka_mnesia:create_table(?TABLE, [{type, ordered_set},
+                                           {ram_copies, [node()]},
+                                           {record_name, ?TABLE},
+                                           {attributes, record_info(fields, ?TABLE)}
+                                          ]);
+mnesia(copy) ->
+    ok = ekka_mnesia:copy_table(?TABLE, ram_copies).
+
+wait_full_replication(Cluster) ->
+    wait_full_replication(Cluster, 100).
+
+%% Emit a special transaction and wait until all replicants replicate it.
+wait_full_replication(Cluster, Timeout) ->
+    [CoreNode|_] = [N || #{node := N, role := core} <- Cluster],
+    emit_last_transaction(CoreNode),
+    [{ok, _} = ?block_until(#{ ?snk_kind := rlog_import_trans
```

Comment:
This is a situation where `?wait_async_action` would be ideal, but it doesn't allow waiting for multiple events... Maybe emitting a transaction per node is no big deal

Reply:
Another idea would be to put a `make_ref()` to the transaction.